File: benchmark/PHP 7.0/哈希表性能和内存占用情况.php
<?php

class a {
	public $v1 = 1;
	public $v2 = 2;
	public $v3 = 3;
	public $v4 = 4;
	public $v5 = 5;
	public $v6 = 6;
}

class b {
	public $v1 = 1;
	public $v2 = 2;
	public $v3 = 3;
	public $v4 = 4;
	public $v5 = 5;
	public $v6 = 6;
	
	public $v7 = 7;
	public $v8 = 7;
	public $v9 = 7;
	public $v10 = 7;
	public $v11 = 7;
	public $v12 = 7;
}

$starttime = microtime(true);

$a = array();
for ($i = 0; $i < 1000*1000; $i++)
	$a[$i] = new a();
echo microtime(true)-$starttime."\n";
for ($i = 0; $i < 1000*1000; $i+=2)
	unset($a[$i]);
echo microtime(true)-$starttime."\n";
$b = array();
for ($i = 0; $i < 1000*1000; $i+=2)
	$b[$i] = new b();
	
echo microtime(true)-$starttime."\n";

File: benchmark/CZL 1.0/哈希表性能和内存占用情况.c
class a
{
	var v1=1, v2=2, v3=3, v4=4, v5=5, v6=6
}

class b
{
	var v1=1, v2=2, v3=3, v4=4, v5=5, v6=6
	var v7=1, v8=2, v9=3, v10=4, v11=5, v12=6
}

main
{
	setrank(0)
	
	a = {}
	for i in (0, 1000*1000)
		a[i] = new a
	echo(clock())
	
	for i in (0, 1000*1000, 2)
		i => a
	echo(clock())
	
	b = {}
	for i in (0, 1000*1000, 2)
		b[i] = new b
	echo(clock())
}
